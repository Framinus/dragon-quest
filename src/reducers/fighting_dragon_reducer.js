import { FIGHTING_DRAGON } from '../actions';

export default function(state={}, action) {
  switch(action.type) {
    case FIGHTING_DRAGON:
      return action.payload
    default: return state
  }
}
