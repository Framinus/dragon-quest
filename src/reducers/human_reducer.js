import { SPAWN_HUMAN, SAVE_NEW_HUMAN } from '../actions';

export default function (state={}, action) {
  switch(action.type) {
    case SPAWN_HUMAN:
      return action.payload.data.human
    case SAVE_NEW_HUMAN:
      return action.payload
    default: return state;
  }
}
