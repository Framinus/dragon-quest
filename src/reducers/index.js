import { combineReducers } from 'redux';
import DragonsReducer from './reducer_dragons';
import RandomDragonReducer from './reducer_random_dragon';
import EnterFightModeReducer from './reducer_enter_fight';
import FightingDragonReducer from './reducer_fighting_dragon';
import HumanReducer from './reducer_human';
import MergingDragonsReducer from './reducer_merging_dragons';

const rootReducer = combineReducers({
  dragons: DragonsReducer,
  fightMode: EnterFightModeReducer,
  fightingDragon: FightingDragonReducer,
  human: HumanReducer,
  mergingDragons: MergingDragonsReducer,
  randomDragon: RandomDragonReducer,
});

export default rootReducer;
