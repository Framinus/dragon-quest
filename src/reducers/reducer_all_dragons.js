import { GET_DRAGON_LIST } from '../actions';

export default function reducerAllDragons(state=[], action) {
  switch(action.type) {
    case GET_DRAGON_LIST:
      return action.payload
    default: return state;
  }
}
