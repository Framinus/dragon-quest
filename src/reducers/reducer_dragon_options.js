import { DRAGON_BY_ID } from '../actions';

export default function(state={}, action) {
  switch(action.type) {
    case DRAGON_BY_ID:
      return action.payload.data;
    default: return state
  }
}
