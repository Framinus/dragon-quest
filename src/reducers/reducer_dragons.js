import { ADD_DRAGON } from '../actions';

export default function(state = [], action) {
  switch(action.type) {
    case ADD_DRAGON:
      return [...state, action.payload];
    default:
      return state;
  }
}
