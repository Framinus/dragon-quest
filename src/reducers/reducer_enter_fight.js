import { ENTER_FIGHT } from '../actions';

export default function(state={}, action) {
  switch(action.type) {
    case ENTER_FIGHT:
      return action.payload
    default: return state
  }
}
