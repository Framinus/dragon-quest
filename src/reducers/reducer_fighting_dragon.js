import { CLEAR_FIGHTING_DRAGON, FIGHTING_DRAGON, UPDATE_DRAGON_HP } from '../actions';

export default function(state={}, action) {
  switch(action.type) {
    case CLEAR_FIGHTING_DRAGON:
      return {}
    case FIGHTING_DRAGON:
      return action.payload
    case UPDATE_DRAGON_HP:
      return { ...state, currenthp: action.payload }
    default: return state
  }
}
