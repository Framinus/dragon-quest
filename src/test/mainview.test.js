import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import promise from 'redux-promise';
import { Provider } from 'react-redux';
import enzyme, { shallow, mount, render } from 'enzyme';
import { createStore, applyMiddleware, compose } from 'redux';
import renderer from 'react-test-renderer';
import toJSON from 'enzyme-to-json';
import MainView from '../components/mainView';
import reducers from '../reducers';

enzyme.configure({ adapter: new Adapter() });

const createStoreWithMiddleware = createStore(
  reducers,
  compose(
    applyMiddleware(promise))
  );

describe('MainView component', () => {

  const wrapper = shallow(<MainView />);

  it('should be selectable by "main-view-container"', () => {
    expect(wrapper.is('.main-view-container')).toBe(true);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Provider store={createStoreWithMiddleware} >
        <MainView />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

})
