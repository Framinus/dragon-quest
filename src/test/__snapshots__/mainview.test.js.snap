// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView component should match its empty snapshot 1`] = `
<div
  className="main-view-container"
>
  <div
    className="dragon-collection-description"
  >
    <div
      className="control-btns"
    >
      <span>
        <button
          className="btn btn-success"
          id="dragon-call-btn"
          onClick={undefined}
        >
          Call Dragon
        </button>
      </span>
      <span>
        <button
          className="btn btn-info"
          id="dragon-merge-btn"
          onClick={undefined}
        >
          Merge Mode
        </button>
      </span>
    </div>
  </div>
  <div
    className="carousel"
  >
    <div>
      <div>
        <h4>
          No dragons here!
        </h4>
      </div>
    </div>
  </div>
</div>
`;
